src/github.com/matrix-org/dendrite/clientapi/auth/auth.go

```diff
@@ -40,6 +40,8 @@ var deviceIDByteLength = 6
 type DeviceDatabase interface {
 	// Look up the device matching the given access token.
 	GetDeviceByAccessToken(ctx context.Context, token string) (*authtypes.Device, error)
+	// Look up the device matching the given id.
+	GetDeviceByID(ctx context.Context, id string) (*authtypes.Device, error)
```

Comment:
I don't think we need it in this interface.

Reply:
Right, should've figured that since GetDevicesByLocalpart works nicely without being included.